demos/fleet_provisioning/fleet_provisioning_with_csr/fleet_provisioning_csr_demo.c

```diff
@@ -0,0 +1,874 @@
+        {
+            /* Publish the RegisterThing request. */
+            PublishToTopic( FP_CBOR_REGISTER_PUBLISH_TOPIC( PROVISIONING_TEMPLATE_NAME ),
+                            FP_CBOR_REGISTER_PUBLISH_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ),
+                            payloadBuffer,
+                            payloadLength );
+
+            if( status == false )
+            {
+                LogError( ( "Failed to publish to fleet provisioning topic: %.*s.",
+                            FP_CBOR_REGISTER_PUBLISH_LENGTH( PROVISIONING_TEMPLATE_NAME_LENGTH ),
+                            FP_CBOR_REGISTER_PUBLISH_TOPIC( PROVISIONING_TEMPLATE_NAME ) ) );
+            }
+        }
+
+        if( status == true )
+        {
+            /* Get the response to the RegisterThing request. */
+            status = waitForResponse();
+        }
+
+        if( status == true )
+        {
+            /* Extract the Thing name from the response. */
+            thingNameLength = MAX_THING_NAME_LENGTH;
+            status = parseRegisterThingResponse( payloadBuffer,
+                                                 payloadLength,
+                                                 thingName,
+                                                 &thingNameLength );
+
+            if( status == true )
+            {
+                LogInfo( ( "Received Thing name: %.*s", ( int ) thingNameLength, thingName ) );
+            }
+        }
+
+        if( status == true )
+        {
+            /* Unsubscribe from the RegisterThing topics. */
+            unsubscribeFromRegisterThingResponseTopics();
+        }
+
+        /**** Disconnect from AWS IoT Core ************************************/
+
+        /* As we have completed the provisioning workflow, we disconnect from
+         * the connection using the provisioning claim credentials. */
+        if( connectionEstablished == true )
+        {
+            DisconnectMqttSession();
+            connectionEstablished = false;
+        }
+
+        /**** Connect to AWS IoT Core with provisioned certificate ************/
+
+        if( status == true )
+        {
+            LogInfo( ( "Establishing MQTT session with provisioned certificate..." ) );
+            status = EstablishMqttSession( provisioningPublishCallback,
+                                           PROVISIONING_CERT_PATH,
+                                           PROVISIONING_PRIVATE_KEY_PATH );
+
+            if( status != true )
+            {
+                LogError( ( "Failed to establish MQTT session." ) );
+            }
+            else
+            {
+                connectionEstablished = true;
+            }
+        }
+
+        /**** Finish **********************************************************/
+
+        if( connectionEstablished == true )
+        {
+            /* Close the connection. */
+            DisconnectMqttSession();
+            connectionEstablished = false;
+        }
+
+        if( status == true )
```

Comment:
The use of `status` and `existStatus` seems redundant, the return logic of this function can simply have `return (status == true)? EXIT_SUCCESS: EXIT_FAILURE;`